op-node/rollup/engine/engine_controller.go

```diff
@@ -105,7 +103,16 @@ type EngineController struct {
 	SyncDeriver SyncDeriver
 }
 
-func NewEngineController(engine ExecEngine, log log.Logger, metrics ECMetrics,
+// EngControllerInterface provides direct calls into the EngineController that
+// external components can use instead of emitting events.
+type EngControllerInterface interface {
+	// TryUpdatePendingSafe updates the pending safe head if the new reference is newer
+	TryUpdatePendingSafe(ctx context.Context, ref eth.L2BlockRef, concluding bool, source eth.L1BlockRef)
+	// TryUpdateLocalSafe updates the local safe head if the new reference is newer and concluding
+	TryUpdateLocalSafe(ctx context.Context, ref eth.L2BlockRef, concluding bool, source eth.L1BlockRef)
+}
+
+func NewEngineControllerWithEventFields(ctx context.Context, engine ExecEngine, log log.Logger, m opmetrics.Metricer,
```

Comment:
What is the rename from `NewEngineController` to `NewEngineControllerWithEventFields` intending to communicate? Keeping the name as `NewEngineController` may be preferable and help clarify the diff.